src/components/forms/ComboBox/ComboBox.test.tsx

```diff
@@ -0,0 +1,1111 @@
+    })
+
+    it('hides options list when clicking away', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-input'))
+      fireEvent.blur(getByTestId('combo-box-input'))
+
+      expect(getByTestId('combo-box-input')).toHaveAttribute(
+        'aria-expanded',
+        'false'
+      )
+      expect(getByTestId('combo-box-option-list')).not.toBeVisible()
+    })
+
+    it('shows and hides options list when toggle is clicked', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-toggle'))
+
+      expect(getByTestId('combo-box-input')).toHaveAttribute(
+        'aria-expanded',
+        'true'
+      )
+      expect(getByTestId('combo-box-option-list')).toBeVisible()
+
+      fireEvent.click(getByTestId('combo-box-toggle'))
+
+      expect(getByTestId('combo-box-input')).toHaveAttribute(
+        'aria-expanded',
+        'false'
+      )
+      expect(getByTestId('combo-box-option-list')).not.toBeVisible()
+    })
+
+    it('selects an item by clicking on an option', () => {
+      const onChange = jest.fn()
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={onChange}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-toggle'))
+      fireEvent.click(getByTestId('combo-box-option-apple'))
+
+      expect(onChange).toHaveBeenCalledWith(fruitOptions[0].value)
+    })
+
+    it('persists input text if dropdown is closed and open without selection', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      const input = getByTestId('combo-box-input')
+      userEvent.type(input, 'yu')
+
+      userEvent.click(getByTestId('combo-box-toggle'))
+      expect(input).toHaveValue('yu')
+
+      userEvent.click(getByTestId('combo-box-toggle'))
+      expect(input).toHaveValue('yu')
+    })
+
+    it('updates input with item selected on click', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      const input = getByTestId('combo-box-input')
+      userEvent.type(input, 'yu')
+      userEvent.click(getByTestId('combo-box-option-yuzu'))
+
+      expect(getByTestId('combo-box-input')).toHaveValue('Yuzu')
```

Comment:
It might be worth testing that the `onChange` handler was called with the value here